### PR TITLE
CUMULUS-151

### PR DESCRIPTION
**Pull request by kkelly51**
Adding validation for JSON schemas

### CODE REVIEW

File: packages/sled/index.js [hunk 1]
```diff
@@ -61,6 +61,28 @@ function getNestedHandler(handlerString) {
...
+ * @param {String} input The input, config, or output to be validated
+ * @param {String} schemaFile The relative path to the schema
+ * @returns {Promise} A Promise resolving to true if Input is valid or rejecting if it errors
+ */
+function validateMessage(input, schemaFile) {
+  if (input && schemaFile) {
+    return readJsonFile(schemaFile)
+      .then((schema) => {
+        const ajv = new Ajv();
+        const schemaValidator = ajv.compile(schema);
```
Comment (yjpa7145) [inline packages/sled/index.js]: This can be removed if you use `const valid = ajv.validate(schema, input);`

File: packages/sled/index.js [hunk 2]
```diff
@@ -61,6 +61,28 @@ function getNestedHandler(handlerString) {
...
+        const valid = schemaValidator(input);
+        if (!valid) {
+          throw Error('Validation Error', schemaValidator.errors);
+        }
+        return valid;
```
Comment (yjpa7145) [inline packages/sled/index.js]: `return true;` would make it more clear that this is returning a boolean.  (The ajv docs don't make it totally clear what "valid" is set to if there were no errors.)

File: packages/sled/index.js (hunk 1 above)
Comment (kkelly51) [reply packages/sled/index.js]: I liked having the object schemaValidator because I could pull the error details out of it 
schemaValidator.errors

Comment (yjpa7145) [reply packages/sled/index.js]: This is the example in the docs:

```
// ...
var valid = ajv.validate(schema, data);
if (!valid) console.log(ajv.errors);
// ...
```

It looks like you can pull the errors from `ajv` in that case.  I don't really have a strong opinion on this one.  If you're happy then I'm happy.

**Pull request merged**